File: strings/387. First Unique Character in a String/firstUniqChar.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let count = {};

    for (let c of s) {
        if (c in count) {
            count[c]++;
        } else {
            count[c] = 1;
        }
    }

    for (let i = 0; i < s.length; i++) {
        if (count[s[i]] === 1) {
            return i;
        }
    }

    return -1;
};

File: strings/387. First Unique Character in a String/firstUniqChar.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        count = {}
        for i in range(len(s)):
            if s[i] in count:
                count[s[i]] += 1
            else:
                count[s[i]] = 1
        
        for i in range(len(s)):
            if count[s[i]] == 1:
                return i
        
        return -1